app/jobs/generate_referral_report_job.rb

```diff
@@ -0,0 +1,28 @@
+class GenerateReferralReportJob < ApplicationJob
+  queue_as :scheduler
+
+  def perform(publisher_id:, referral_codes:, start_date:, end_date:, interval:, break_down_by_country:)
+    report_csv = Promo::RegistrationStatsReportGenerator.new(
+      referral_codes: referral_codes,
+      start_date: start_date,
+      end_date: end_date,
+      reporting_interval: interval,
+      is_geo: break_down_by_country
+    ).perform
+
+    filename = "tmp/#{Time.now}_referral_report_#{start_date}-#{end_date}.csv"
+    File.open(filename, "w") { |f| f.write(report_csv) }
```

Comment:
I don't really follow how this file gets stored. It gets stored on worker dyno which isn't shared with the web dynos. It would make sense to store the CSV in S3.